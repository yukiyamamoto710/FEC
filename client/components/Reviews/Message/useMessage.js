import { useState } from 'react';

export default function useMessage() {
  const [isClick, setIsClick] = useState(false);

  const handleClickText = () => {
    setIsClick(!isClick);
  };

  const handleKeyPressText = () => {
    setIsClick(!isClick);
  };

  return { isClick, handleClickText, handleKeyPressText };
}
