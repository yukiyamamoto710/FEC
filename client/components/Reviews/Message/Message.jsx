import React from 'react';
import PropTypes from 'prop-types';
import useMessage from './useMessage';

const Message = (props) => {
  const { body } = props;
  const {
    isClick,
    handleClickText,
    handleKeyPressText,
  } = useMessage();

  if (isClick) {
    return (
      <div
        data-testid="textShow"
        className="msgBody1"
        role="button"
        tabIndex={0}
        onKeyPress={handleKeyPressText}
        onClick={handleClickText}
      >
        {body}
      </div>
    );
  }
  return (
    <div
      data-testid="textHidden"
      className="msgBody"
      role="button"
      tabIndex={0}
      onKeyPress={handleKeyPressText}
      onClick={handleClickText}
    >
      {body}
    </div>
  );
};

Message.propTypes = {
  body: PropTypes.string,
};

Message.defaultProps = {
  body: '',
};

export default Message;
