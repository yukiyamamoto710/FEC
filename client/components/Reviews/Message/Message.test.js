/**
* @jest-environment jsdom
*/
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  cleanup,
  fireEvent,
  screen,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import Message from './Message';

afterEach(cleanup);

it('render text correct', () => {
  const { getByTestId } = render(<Message body="kenneth@@" />);
  expect(getByTestId('textHidden')).toBeInTheDocument();
  expect(screen.queryByText('kenneth@@')).toBeInTheDocument();
});

it('render text correct after click', () => {
  const { getByTestId } = render(<Message body="kenneth@@" />);
  fireEvent.click(getByTestId('textHidden'));
  expect(getByTestId('textShow')).toBeInTheDocument();
  fireEvent.click(getByTestId('textShow'));
  expect(getByTestId('textHidden')).toBeInTheDocument();
  fireEvent.keyPress(getByTestId('textHidden'), { key: 'Enter', code: 13, charCode: 13 });
  expect(getByTestId('textShow')).toBeInTheDocument();
  fireEvent.keyPress(getByTestId('textShow'), { key: 'Enter', code: 13, charCode: 13 });
  expect(getByTestId('textHidden')).toBeInTheDocument();
});

it('matches snapShot', () => {
  const tree = renderer.create(
    <Message body="kenneth@@" />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
