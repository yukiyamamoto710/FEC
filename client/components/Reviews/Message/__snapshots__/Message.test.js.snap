// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="msgBody"
  data-testid="textHidden"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  kenneth@@
</div>
`;
